<?xml version="1.0"?>
<launch>
  <!-- Arguments -->
  <arg name="cmd_vel_topic" default="cmd_vel" />
  <arg name="odom_topic" default="odom" />
  <arg name="move_forward_only" default="false"/>

  <!-- move_base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
    <rosparam file="$(find navigtion_pkg)/parameter/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find navigtion_pkg)/parameter/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find navigtion_pkg)/parameter/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find navigtion_pkg)/parameter/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find navigtion_pkg)/parameter/move_base_params.yaml" command="load" />
    <rosparam file="$(find navigtion_pkg)/parameter/dwa_local_planner_params.yaml" command="load" />
    <!--param name="global_costmap/global_frame" value="/map"/-->
    <param name="global_costmap/robot_base_frame" value="robot1/base_link"/>
    <param name="local_costmap/robot_base_frame" value="robot1/base_link"/>
    <param name="local_costmap/global_frame" value="robot1/odom"/>
    <param name="DWAPlannerROS/min_vel_x" value="0.0" if="$(arg move_forward_only)" />
    <!-- remap from="map" to="/map"/-->
    <!-- remap from="cmd_vel" to="$(arg cmd_vel_topic)"/ -->
    <!-- remap from="odom" to="$(arg odom_topic)"/ -->
  </node>
</launch>
